### DISCUSSION TITLE
preload() Twig function and responsive images

### DISCUSSION THREAD

**Discussion by CyrilKrylatov** [symfony/symfony#50569] [Q&A] [answered]
Hello everyone, 

I have this question about the [preload()](https://symfony.com/doc/current/web_link.html) function and responsives images.

When I want to use `link[preload]` HTML tag to preload a responsive image, I use it like this:

```twig
<link rel="preload" href="{{ user.getCoverWebPath|imagine_filter('user_cover_320w') }}" as="image" media="(max-width: 320px)">
<link rel="preload" href="{{ user.getCoverWebPath|imagine_filter('user_cover_480w') }}" as="image" media="(min-width: 321px) and (max-width: 480px)">
<link rel="preload" href="{{ user.getCoverWebPath|imagine_filter('user_cover_640w') }}" as="image" media="(min-width: 481px) and (max-width: 640px)">
<link rel="preload" href="{{ user.getCoverWebPath|imagine_filter('user_cover_960w') }}" as="image" media="(min-width: 641px)">
```

Let's say I want to use the `preload()` function to send the image right in the HTTP Headers, I'd do this:

```twig
<link rel="preload" href="{{ preload(user.getCoverWebPath|imagine_filter('user_cover_320w'), { as: 'image' }) }}" as="image" media="(max-width: 320px)">
<link rel="preload" href="{{ preload(user.getCoverWebPath|imagine_filter('user_cover_480w'), { as: 'image' }) }}" as="image" media="(min-width: 321px) and (max-width: 480px)">
<link rel="preload" href="{{ preload(user.getCoverWebPath|imagine_filter('user_cover_640w'), { as: 'image' })  }}" as="image" media="(min-width: 481px) and (max-width: 640px)">
<link rel="preload" href="{{ preload(user.getCoverWebPath|imagine_filter('user_cover_960w'), { as: 'image' })  }}" as="image" media="(min-width: 641px)">
```

But in this last case, the four images will be preloaded since the HTTP Headers don't manage the `media` attribute. So if I send to it 4 images, it will preload the 4 images. Heh.

So my question is: how can I use the `preload()` with responsives images without preloading them all?

Thank you for your time answer and your time!

Cheers,

Cyril

**Comment by MatTheCat** [comment, marked as answer]
`Link` header parameters are supposed to be the same than the `<link>` element attributes, so you should be able to add `media` to `preload`’s attributes.

**Reply by CyrilKrylatov** [reply]
Oh. It's very true. Thank you!
